deluge/core/rpcserver.py

```diff
@@ -530,10 +530,10 @@ def generate_ssl_keys():
     """
     This method generates a new SSL key/cert.
     """
-    digest = "md5"
+    digest = "sha256"
     # Generate key pair
     pkey = crypto.PKey()
-    pkey.generate_key(crypto.TYPE_RSA, 1024)
+    pkey.generate_key(crypto.TYPE_RSA, 3072)
```

Comment:
Short answer: Yes.

Longer answer: [Wikipedia](https://en.wikipedia.org/wiki/Key_size#Asymmetric_algorithm_key_lengths) suggests 3072 is sufficient for 128bit symmetric parity. 2048 is claimed to be sufficient until 2030 (i.e. a change would need to be made in 2027 to bump the strength again), but given that 3072 is not significantly larger and offers reasonable protection against non-quantum attackers beyond 2030, is there any reason to use a shorter key length?